tools/setup/emoji/emoji_names.py

```diff
@@ -1,1580 +1,2055 @@
-    "1f3ba": {"canonical_name": "trumpet", "aliases": []},
-    "1f3b8": {"canonical_name": "guitar", "aliases": []},
-    "1f3bb": {"canonical_name": "violin", "aliases": []},
-    # dice seems more useful
-    "1f3b2": {"canonical_name": "dice", "aliases": ["die"]},
-    # direct_hit from gemoji/Unicode, and seems more useful. bulls_eye seemed
-    # like a reasonable addition
-    "1f3af": {"canonical_name": "direct_hit", "aliases": ["darts", "bulls_eye"]},
-    # strike seemed more useful than bowling
-    "1f3b3": {"canonical_name": "strike", "aliases": ["bowling"]},
-    "1f3ae": {"canonical_name": "video_game", "aliases": []},
-    # gambling seemed more useful than slot_machine
-    "1f3b0": {"canonical_name": "slot_machine", "aliases": []},
-    # the Google emoji for this is not red
-    "1f697": {"canonical_name": "car", "aliases": []},
-    # rideshare seems like a reasonable addition
-    "1f695": {"canonical_name": "taxi", "aliases": ["rideshare"]},
-    # the Google emoji for this is not blue. recreational_vehicle is from
-    # gemoji/Unicode, jeep seemed like a good addition
-    "1f699": {"canonical_name": "recreational_vehicle", "aliases": ["jeep"]},
-    # school_bus seemed like a reasonable addition, even though the twitter
-    # glyph for this doesn't really look like a school bus
+    "1f636": {"canonical_name": "speechless", "aliases": ["no_mouth", "blank", "poker_face"]},
+    "1f637": {"canonical_name": "mask", "aliases": []},
+    "1f638": {"canonical_name": "smile_cat", "aliases": []},
+    "1f639": {"canonical_name": "joy_cat", "aliases": []},
+    "1f63a": {"canonical_name": "smiley_cat", "aliases": []},
+    "1f63b": {"canonical_name": "heart_eyes_cat", "aliases": []},
+    "1f63c": {"canonical_name": "smirk_cat", "aliases": ["smug_cat"]},
+    "1f63d": {"canonical_name": "kissing_cat", "aliases": []},
+    "1f63e": {"canonical_name": "angry_cat", "aliases": ["pouting_cat"]},
+    "1f63f": {"canonical_name": "crying_cat", "aliases": []},
+    "1f640": {"canonical_name": "scream_cat", "aliases": ["weary_cat"]},
+    "1f641": {"canonical_name": "frown", "aliases": ["slight_frown"]},
+    "1f642": {"canonical_name": "smile", "aliases": []},
+    "1f643": {"canonical_name": "upside_down", "aliases": ["oops"]},
+    "1f644": {"canonical_name": "rolling_eyes", "aliases": []},
+    "1f645-200d-2640": {"canonical_name": "woman gesturing NO", "aliases": []},
+    "1f645-200d-2642": {"canonical_name": "man gesturing NO", "aliases": []},
+    "1f645": {"canonical_name": "no_signal", "aliases": ["nope"]},
+    "1f646-200d-2640": {"canonical_name": "woman gesturing OK", "aliases": []},
+    "1f646-200d-2642": {"canonical_name": "man gesturing OK", "aliases": []},
+    "1f646": {"canonical_name": "ok_signal", "aliases": []},
+    "1f647-200d-2640": {"canonical_name": "woman bowing", "aliases": []},
+    "1f647-200d-2642": {"canonical_name": "man bowing", "aliases": []},
+    "1f647": {"canonical_name": "bow", "aliases": []},
```

Comment:
Hmm. How does it work out when there are spaces in the `canonical_name` like this? Does typeahead give you e.g. `:man bowing:`? Does that get rendered successfully as the intended emoji?

For as long as we have the model that emoji have names and the names appear in message Markdown, I think we probably want to keep the names in the lowercase-underscores style. That should mostly be a matter of replacing `' '` with `'_'`, and perhaps a few other replacements.

Reply:
We already have spaces! Check out `:stuck out tongue:` .... oh I see, it shows it without underscores in the search view but uses underscores when you complete typing it (and in `emoji_names.py`). I'll play around with fixing that!